core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/lateacceptance/DiversifiedLateAcceptanceAcceptor.java

```diff
@@ -0,0 +1,113 @@
+
+    // The worst score in the late elements list
+    protected Score<?> lateWorse;
+    // Number of occurrences of lateWorse in the late elements
+    protected int lateWorseOccurrences = -1;
+
+    protected int lateAcceptanceSize = -1;
+
+    protected Score<?>[] previousScores;
+    protected int lateScoreIndex = -1;
+
+    public void setLateAcceptanceSize(int lateAcceptanceSize) {
+        this.lateAcceptanceSize = lateAcceptanceSize;
+    }
+
+    // ************************************************************************
+    // Worker methods
+    // ************************************************************************
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
+        super.phaseStarted(phaseScope);
+        validate();
+        previousScores = new Score[lateAcceptanceSize];
+        var initialScore = phaseScope.getBestScore();
+        Arrays.fill(previousScores, initialScore);
+        lateScoreIndex = 0;
+        lateWorseOccurrences = lateAcceptanceSize;
+        lateWorse = initialScore;
+    }
+
+    private void validate() {
+        if (lateAcceptanceSize <= 0) {
+            throw new IllegalArgumentException(
+                    "The lateAcceptanceSize (%d) cannot be negative or zero.".formatted(lateAcceptanceSize));
+        }
+    }
+
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public boolean isAccepted(LocalSearchMoveScope<Solution_> moveScope) {
+        // The acceptance and replacement strategies are based on the work:
+        // Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar
+        var lateScore = previousScores[lateScoreIndex];
+        var moveScore = moveScope.getScore();
+        var current = (Score) moveScope.getStepScope().getPhaseScope().getLastCompletedStepScope().getScore();
+        var previous = current;
+        var accept = moveScore.compareTo(current) == 0 || moveScore.compareTo(lateWorse) > 0;
+        if (accept) {
+            current = moveScore;
+        }
+        // Improves the diversification to allow the next iterations to find a better solution
+        var currentScoreWorse = current.compareTo(lateScore) < 0;
+        // Improves the intensification but avoids replacing values when the search falls into a plateau or local minima
+        var currentScoreBetter = current.compareTo(lateScore) > 0 && current.compareTo(previous) > 0;
+        if (currentScoreWorse || currentScoreBetter) {
+            updateLateScore(current);
+        }
+        lateScoreIndex = (lateScoreIndex + 1) % lateAcceptanceSize;
+        return accept;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void updateLateScore(Score newScore) {
+        var newScoreWorse = newScore.compareTo(lateWorse) < 0;
+        var newScoreEqual = newScore.compareTo(lateWorse) == 0;
```

Comment:
Please reuse the comparison. This is called relatively frequently, it needs to be efficient.